python/lsst/daf/butler/registry/obscore/_records.py

```diff
@@ -189,11 +191,15 @@ def __call__(self, ref: DatasetRef, context: SqlQueryContext) -> Optional[Record
                 self._visit_records(dimension_record, record)
 
         # ask each plugin for its values to add to a record.
-        for plugin in self.spatial_plugins:
-            assert ref.id is not None, "Dataset ID must be defined"
-            plugin_record = plugin.make_records(ref.id, region)
-            if plugin_record is not None:
-                record.update(plugin_record)
+        try:
+            plugin_records = self.make_spatial_records(region)
+        except RegionTypeError as exc:
+            warnings.warn(
```

Comment:
Philosophically do we want this to be a warning or a fatal/critical log message? (also above)

Reply:
I prefer warning just in case in the future someone wants to define visits with odd region types, they could at least suppress the messages. Fatal message would probably mean to some people that things are failing badly, while we are only ignoring that region data, but we fill the rest of the obscore record correctly. And I hope we'll never actually see this message.